packages/synapse-react-client/src/components/GoalsV2/GoalsV2.tsx

```diff
@@ -0,0 +1,235 @@
+    ExpectedColumns.SUMMARY,
+    queryResultBundle,
+  )
+  const linkColumnIndex = getFieldIndex(ExpectedColumns.LINK, queryResultBundle)
+
+  return (
+    <Box
+      sx={{
+        flex: '1 1 auto',
+        backgroundColor: 'rgba(0, 128, 0, 0.05)',
+        zIndex: 0,
+        height: '560px',
+        padding: '80px',
+      }}
+    >
+      <Box
+        sx={{
+          flex: '1 1 auto',
+          zIndex: 1,
+          textAlign: 'center',
+          paddingBottom: '30px',
+        }}
+      >
+        <Typography
+          variant="headline1"
+          style={{ color: 'black', fontFamily: 'Merriweather', fontSize: '32' }}
+          sx={{
+            pt: '30px',
+            mb: '10px',
+            mx: 'auto',
+            width: 'max-content',
+            borderTop: '3px solid rgba(128, 128, 128, 0.25)',
+          }}
+        >
+          What's in the Portal?
+        </Typography>
+        <Link
+          href="https://eliteportal.synapse.org/Explore/Data"
+          target="_blank"
+          rel="noopener noreferrer"
+          fontFamily="Lato"
+          fontSize="18"
+          fontStyle="semi-bold"
+        >
+          <Button
+            variant="contained"
+            sx={{
+              backgroundColor: '#5BA998',
+              border: '1px solid white',
+              boxShadow: 'none',
+              '&:hover': {
+                backgroundColor: darken('#5BA998', 0.05),
+                boxShadow: 'none',
+              },
+            }}
+          >
+            Start Exploring Data
+          </Button>
+        </Link>
+      </Box>
+      {error && <ErrorBanner error={error} />}
+      <div className={`GoalsV2${showDesktop ? '__Desktop' : ''}`}>
+        {queryResultBundle?.queryResult!.queryResults.rows.map((el, index) => {
+          const values = el.values as string[]
+          if (values.some(value => value === null)) {
+            // We cast values above assuming there are no null values, emit a warning just in case.
+            console.warn('Row has null value(s) when no nulls expected')
+          }
+          const tableId =
+            tableIdColumnIndex > -1 ? values[tableIdColumnIndex] : undefined
+          let countSql
+          if (countSqlColumnIndex > -1 && values[countSqlColumnIndex]) {
+            countSql = values[countSqlColumnIndex]
+          } else if (tableId) {
+            countSql = `SELECT * FROM ${tableId}`
+          }
+          const title = values[titleColumnIndex]
+          const summary = values[summaryColumnIndex]
+          const link = values[linkColumnIndex]
+          // assume that we recieve assets in order of rows and there is an asset for each item
+          // can revisit if this isn't the case.
+          const asset = assets?.[index] ?? ''
+          const goalsV2DataProps: GoalsV2DataProps = {
+            countSql,
+            title,
+            summary,
+            link,
+            asset,
+          }
```

Comment:
This is a lot of logic in the returned JSX. Suggest creating an array of `goalsV2DataProps` outside of the `return` and then calling `goalsV2DataPropsArray.map` here